core/src/repair/serve_repair.rs

```diff
@@ -1363,6 +1364,36 @@ pub(crate) fn get_repair_protocol(_: ClusterType) -> Protocol {
     Protocol::UDP
 }
 
+fn deserialize_request<T>(request: &RemoteRequest) -> std::result::Result<T, bincode::Error>
+where
+    T: serde::de::DeserializeOwned,
+{
+    bincode::options()
+        .with_limit(request.bytes.len() as u64)
+        .with_fixint_encoding()
+        .reject_trailing_bytes()
+        .deserialize(&request.bytes)
+}
+
+// Returns true on success.
+fn send_response(
+    packets: PacketBatch,
+    packet_batch_sender: &PacketBatchSender,
+    response_sender: Option<OneShotSender<Vec<Vec<u8>>>>,
+) -> bool {
+    match response_sender {
+        None => packet_batch_sender.send(packets).is_ok(),
+        Some(response_sender) => {
+            let response = packets
+                .iter()
+                .filter_map(|packet| packet.data(..))
+                .map(Vec::from)
```

Comment:
Is it possible to get rid of this map and collect by passing the PacketBatch directly?

Reply:
With UDP patch going away, `PacketBatch` would be superfluous and wasteful.
So I would expect earlier code will be updated not to create `PacketBatch` to begin with.